src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -450,6 +460,74 @@ public void initialLoanDueDateOnCreateWithPrexistingRequests()
 
     assertThat("due date is not the recall due date (2 months)",
         storedLoan.getString("dueDate"), is(expectedDueDate));
+  }
+
+  @Test
+  public void changedDueDateAfterRecallingAnItemShouldRespectTenantTimezone()
+    throws InterruptedException,
+    ExecutionException,
+    TimeoutException,
+    MalformedURLException {
+
+    final String stockholmTimeZone = "Europe/Stockholm";
+
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    configClient.create(timezoneConfigurationFor(stockholmTimeZone));
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+      .withName("Can Circulate Rolling With Recalls")
+      .withDescription("Can circulate item With Recalls")
+      .rolling(Period.days(14))
+      .unlimitedRenewals()
+      .renewFromSystemDate()
+      .withRecallsMinimumGuaranteedLoanPeriod(Period.days(5))
+      .withClosedLibraryDueDateManagement(KEEP_THE_CURRENT_DUE_DATE.getValue());
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.inactiveNotice().getId());
+
+    final DateTime loanDate = DateTime.now(DateTimeZone.UTC).minusDays(3);
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(
+      smallAngryPlanet, steve, loanDate);
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    final DateTime requestDate = DateTime.now(DateTimeZone.UTC);
+
+    freezeTime(requestDate);
+
+    requestsFixture.place(
+      new RequestBuilder()
+        .recall()
+        .forItem(smallAngryPlanet)
+        .by(jessica)
+        .withRequestDate(requestDate)
+        .withPickupServicePoint(requestServicePoint));
+
+    final JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    assertThat("due date is the original date",
+      storedLoan.getString("dueDate"), not(originalDueDate));
+
+    final DateTime expectedDueDate =
+      loanDate.toLocalDate()
+        .toDateTime(END_OF_A_DAY, DateTimeZone.forID(stockholmTimeZone));
+
+    assertThat("due date should be end of the day, 5 days from request date",
```

Comment:
```suggestion
    assertThat("due date should be end of the day, 5 days from loan date",
```

Reply:
@dmytrotur The rules defined in [CIRC-203](https://issues.folio.org/browse/CIRC-203) define the change as
`today's date-time plus the recall return interval = Recall due date-time (RD)`

Where I interpret today's date as the request date, because it is in the context of when the request has just been created.

Does that help make the comment make more sense?

Reply:
Thank you, that is a good spot, I will correct that it in an upcoming pull request